### ISSUE TITLE
DPO Loss approaches 0 too fast

### ISSUE THREAD

**Issue by kangqi-ni** [huggingface/trl#894]
I used a custom dataset to train a fine-tuned version of llama model. However, after a few steps (out of more than 1000 steps in total), the loss got to 0 and the rewards (accuracy) got to 1.0. Did everyone know what is the issue?

**Comment by kangqi-ni** [comment]
I am using a vicuña v1.3 based model, but I didn’t set the EOS token myself since I thought it has already been taken care of. In the hf llama2 dpo tutorial, they created the tokenizer this way: 
tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-7b-hf")
tokenizer.pad_token = tokenizer.eos_token

Do you suggest that I set the tokenizer.eos_token myself and then add this eos_token to the training data?

**Comment by yuanenming** [comment]
Hi my loss curve looks like this, but it actually works well
<img width="718" alt="image" src="https://github.com/huggingface/trl/assets/26831266/b57fd8f5-d832-4214-a81d-42eabdd4463d">

**Comment by yuanenming** [comment]
@OneCodeToRuleThemAll  Yes, It is a conversational task. But my preference data is very easy to be classified.

I think if it is easy to distinguish between chosen and reject samples, the loss will be very low. You should take a look at the accuracy pannel, which suggest how offen chosen reward is higher then reject. If it is near 1, you should get a very low loss.

**Comment by tsavage6001** [comment]
I ran into this same error.  Did you find a solution?
